tests/Proto.Cluster.PartitionIdentity.Tests/HandoverSinkTests.cs

```diff
@@ -0,0 +1,245 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="HandoverTests.cs" company="Asynkron AB">
+//      Copyright (C) 2015-2022 Asynkron AB All rights reserved
+// </copyright>
+// -----------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using Xunit;
+
+namespace Proto.Cluster.PartitionIdentity.Tests;
+
+public class HandoverSinkTests
+{
+    private const string TestKind = "some-kind";
+    private int _counter;
+
+    private static readonly Member TestMember1 = new()
+    {
+        Host = "127.0.0.1",
+        Kinds = {TestKind},
+        Id = Guid.NewGuid().ToString(),
+        Port = 50000
+    };
+
+    private static readonly Member TestMember2 = new()
+    {
+        Host = "127.0.0.1",
+        Kinds = {TestKind},
+        Id = Guid.NewGuid().ToString(),
+        Port = 50001
+    };
+
+    [Fact]
+    public void CompletesOnEmptyHandover()
+    {
+        var topology = CreateTopology(TestMember1, TestMember2);
+        var receivedCount = 0;
+        var sink = new Partition.HandoverSink(topology, handover => receivedCount += handover.Actors.Count);
```

Comment:
Is this called concurrently? Interlocked increments might be needed

Reply:
Even if assign is atomic, reading then adding then assigning is not

Reply:
It's only used within actors, so no need for additional synchronization :)